demo/client/src/components/FilesList.js

```diff
@@ -0,0 +1,123 @@
+import React, { Component } from 'react';
+import config from '../config.json'; // eslint-disable-line import/no-unresolved
+
+const demoServer = `${config.ServiceEndpoint}/demo`;
+
+class FilesList extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      loading: true,
+      files: [],
+      links: {},
+    };
+
+    this.generateLink = this.generateLink.bind(this);
+    this.loadFiles = this.loadFiles.bind(this);
+  }
+
+  componentDidMount() {
+    this.loadFiles();
+  }
+
+  loadFiles() {
+    this.setState({
+      loading: true,
+    });
+
+    window
+      .fetch(`${demoServer}/meta`)
+      .then(response => {
+        if (response.ok) {
+          response.json().then(data => {
+            this.setState({
+              loading: false,
+              files: data,
+            });
+          });
+        } else {
+          console.log('Bad response');
```

Comment:
If to leave if/else in the promise chain, I think it's important to always return values. Seems one of the top [rules](https://github.com/xjamundx/eslint-plugin-promise#promise-rules) to follow.

Reply:
What if there's nothing to return? 😄

Reply:
Read the article from your other comment, it's ok like this.

Reply:
The `handleErrors` thingy would be nice indeed :)